### PR TITLE
Adding params to  close session - on hold while talking with backend team

### PR DESCRIPTION
**Pull request by elopezanaya**
# PR Details

### Description
_Include a description of the problem to be solved_

For conversations ended by agent, when calling close chat session endpoint it throws an exception, since the conversation is already closed , its throwing an exception which is logged and being taking in account by monitors.

## Solution Proposed
_Detail what is the solution proposed, include links to design document if required or any other document required to support the solution_

- passing the flag to indentify when a chat was ended by agent to do not call the endpoint , but stll be able to perform cleanup tasks required for new sessions.

### Acceptance criteria
_Define what are the conditions to consider the PR has achieved the intended goal_
## Test cases and evidence

When session was not ended by customer, dont call close session endpoitn

_Include what tests cases were considered, any evidence of testing for future references, to identify any corner cases, etc_

#### calling end chat without params , expected to call close session endpoint

image

#### calling end chat with params, not calling close session endpoint

image

image

- [ ] You have tested all changes in Popout mode
- [ ] You have tested all changes in cross browsers i.e Edge, Chrome, Firefox, Safari and mobile devices(iOS and Android)
- [ ] Your changes are included in the [CHANGELOG](../CHANGE_LOG.md)

## A11y

- [ ] You have run the [Automated Accessibility Check](https://accessibilityinsights.io/docs/en/windows/getstarted/automatedchecks) and have no reported issues

***Please provide justification if any of the validations has been skipped.***

### CODE REVIEW

**Review by xTEddie — CHANGES_REQUESTED** [review]
It's a API flow behavior change. We'd need to discuss with the service team to verify this scenario.

File: src/OmnichannelChatSDK.ts [hunk 1]
```diff
@@ -779,7 +780,7 @@ class OmnichannelChatSDK {
         }
     }
 
-    public async endChat(): Promise<void> {
+    public async endChat(optionalParams?: EndChatOptionalParams): Promise<void> {
```
Comment (xTEddie) [inline src/OmnichannelChatSDK.ts:783]: What would happen of `chatSDK.endChat()` was called?

Should it have `optionsParams` set to `{}` by default to be backward compatible?
